### PR TITLE
CSL Style selection Tab

### PR DESCRIPTION
**Pull request by subhramit**
This PR finishes the UI changes for selection of CSL Styles.
Changes:
Select Style Window divided into two separate tabs: CSL Styles and JStyles
image
image

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by subhramit** [comment]
Moved to draft as preview of selected style is WIP.

**Comment by subhramit** [comment]
Situation so far [issue faced: preview not being rendered]
image

**Comment by Siedlerchr** [comment]
Okay I tested it. 
Several things:
1.  For the list view use CitationStylePreviewLayout as base this makes it easier to work with for later usage and for the creation of the 
2.  For displaying the titles use a ViewModelFactory  see PreviewTab class:
3. ``` new ViewModelListCellFactory<PreviewLayout>()
                .withText(PreviewLayout::getDisplayName)
                .install(availableListView); ```
4. There is no change listener for the selection of a CSL style (see how it's done for the oo style table)
5. Here comes now the reason why we use the PreviewLayout in our list: we can directly set the style to the   previewArticle.setLayout and the  previewBook.setLayout 
6. You also need to add both of the above previews to the previewVbox (maybe rename this and the other node to make ite easier. e.g one is cslPreviewBox and the other jstylePreviewBox

**Comment by subhramit** [comment]
> Okay I tested it. Several things:
> 
> 1. For the list view use CitationStylePreviewLayout as base this makes it easier to work with for later usage and for the creation of the
> 2. For displaying the titles use a ViewModelFactory  see PreviewTab class:
> 3. ```
>                 .withText(PreviewLayout::getDisplayName)
>                 .install(availableListView); ```
>    ```
> 4. There is no change listener for the selection of a CSL style (see how it's done for the oo style table)
> 5. Here comes now the reason why we use the PreviewLayout in our list: we can directly set the style to the   previewArticle.setLayout and the  previewBook.setLayout
> 6. You also need to add both of the above previews to the previewVbox (maybe rename this and the other node to make ite easier. e.g one is cslPreviewBox and the other jstylePreviewBox

Done, let me know if I missed something.

**Comment by subhramit** [comment]
Current situation:
image

**Comment by subhramit** [comment]
The following lines in ```SelectStyleDialogView``` take care of the listener by default, so I did not separately add one for CSL Styles:
image

**Comment by subhramit** [comment]
Failing test: "Deploy Jekyll site to Pages / build (push)". 
```
Error: Get Pages site failed. Please verify that the repository has Pages enabled and configured to build using GitHub Actions, or consider exploring the `enablement` parameter for this action. Error: Not Found - https://docs.github.com/rest/pages/pages#get-a-apiname-pages-site
Error: HttpError: Not Found - https://docs.github.com/rest/pages/pages#get-a-apiname-pages-site
```

On enabling pages:
```Liquid Exception: Liquid syntax error (line 7): Unknown tag 'figure' in docs/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-11-code-into-ide.md```
```
Error:  Logging at level: debug GitHub Pages: github-pages v231 GitHub Pages: jekyll v3.9.5 Theme: jekyll-theme-primer Theme source: /usr/local/bundle/gems/jekyll-theme-primer-0.6.0 Requiring: jekyll-github-metadata Requiring: jekyll-seo-tag Requiring: jekyll-coffeescript Requiring: jekyll-commonmark-ghpages Requiring: jekyll-gist Requiring: jekyll-github-metadata Requiring: jekyll-paginate Requiring: jekyll-relative-links Requiring: jekyll-optional-front-matter Requiring: jekyll-readme-index Requiring: jekyll-default-layout Requiring: jekyll-titles-from-headings GitHub Metadata: Initializing... Source: /github/workspace/. Destination: /github/workspace/./_site Incremental build: disabled. Enable with --incremental Generating... Generating: JekyllOptionalFrontMatter::Generator finished in 0.005824906 seconds. Generating: JekyllReadmeIndex::Generator finished in 0.054141545 seconds. Generating: Jekyll::Paginate::Pagination finished in 4.939e-06 seconds. GitHub Metadata: Generating for subhramit/jabref GitHub
```

**Comment by Siedlerchr** [comment]
You can ignore the jekyll stuff, from what I see it's some plugin issue that appears within github pages
https://littlecheesecake.medium.com/publish-jekyll-site-to-github-pages-876ce2a8279c

File: src/main/java/org/jabref/gui/openoffice/StyleSelectDialogView.java [hunk 1]
```diff
@@ -140,10 +148,6 @@ private void initialize() {
             previewArticle.setLayout(new TextBasedPreviewLayout(style.getStyle().getReferenceFormat(StandardEntryType.Article)));
             previewBook.setLayout(new TextBasedPreviewLayout(style.getStyle().getReferenceFormat(StandardEntryType.Book)));
         });
-
-        availableListView.setItems(viewModel.getAvailableStyles());
-        searchBox.textProperty().addListener((observable, oldValue, newValue) ->
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/openoffice/StyleSelectDialogView.java]: You have removed the search feature....

Comment (subhramit) [reply src/main/java/org/jabref/gui/openoffice/StyleSelectDialogView.java]: aye, damn. Fixed.

**Pull request closed without merging**